regression/run/references/main.f

```diff
@@ -58,10 +59,24 @@ subroutine test_arraywrapper
     call assert_true(associated(arr4, arr))
     call assert_equals(10, size(arr4))
 
-    call arrinst%fetch_array(arr5)
+    call arrinst%fetch_array_ptr(arr5)
     call assert_true(associated(arr5, arr))
     call assert_equals(10, size(arr5))
-    
+
+    call arrinst%fetch_array_ref(arr6)
+    call assert_true(associated(arr6, arr))
+    call assert_equals(10, size(arr6))
```

Comment:
It might be helpful here to add a loop that accesses, and does something with the array items.
E.g. sets them, checks them, prints them, ....

Reply:
I added some checks that modify the array from Fortran/Python then sums the array in the class to make sure the correct address is returned.  All the other tests use `associated` to compare the the pointers (This Fortran intrinsic checks to see if the arrays are 'associated' with each other)